tests/integration/e2e/helpers.py

```diff
@@ -120,3 +121,107 @@ async def fetch_action_stop_process(unit: Unit) -> Dict:
+
+
+def create_topic(model_full_name: str, app_name: str, topic: str) -> None:
+    """Helper to create a topic.
+
+    Args:
+        model_full_name: Juju model
+        app_name: Kafka app name in the Juju model
+        topic: the desired topic to configure
+    """
+    if SUBSTRATE == "k8s":
+        container_arg = "--container kafka"
+        sudo_arg = ""
+        bin_cmd = "/opt/kafka/bin/kafka-topics.sh"
+        config_file = "/etc/kafka/client.properties"
+    else:
+        container_arg = ""
+        sudo_arg = "sudo -i"
+        bin_cmd = "charmed-kafka.topics"
+        config_file = "/var/snap/charmed-kafka/current/etc/kafka/client.properties"
+
+    try:
+        check_output(
+            f"JUJU_MODEL={model_full_name} juju ssh {container_arg} {app_name}/0 {sudo_arg} '{bin_cmd} --create --topic {topic} --bootstrap-server localhost:19092 "
+            f"--command-config {config_file}'",
+            stderr=STDOUT,
+            shell=True,
+            universal_newlines=True,
+        )
+
+    except CalledProcessError as e:
+        logger.error(f"command '{e.cmd}' return with error (code {e.returncode}): {e.output}")
+        raise
+
+
+def write_topic_message_size_config(
+    model_full_name: str, app_name: str, topic: str, size: int
+) -> None:
+    """Helper to configure a topic's message max size.
+
+    Args:
+        model_full_name: Juju model
+        app_name: Kafka app name in the Juju model
+        topic: the desired topic to configure
+        size: the maximal message size in bytes
+    """
+    if SUBSTRATE == "k8s":
+        container_arg = "--container kafka"
+        sudo_arg = ""
+        bin_cmd = "/opt/kafka/bin/kafka-configs.sh"
+        config_file = "/etc/kafka/client.properties"
+    else:
+        container_arg = ""
+        sudo_arg = "sudo -i"
+        bin_cmd = "charmed-kafka.configs"
+        config_file = "/var/snap/charmed-kafka/current/etc/kafka/client.properties"
+
+    try:
+        result = check_output(
+            f"JUJU_MODEL={model_full_name} juju ssh {container_arg} {app_name}/0 {sudo_arg} '{bin_cmd} --bootstrap-server localhost:19092 "
+            f"--entity-type topics --entity-name {topic} --alter --add-config max.message.bytes={size} --command-config {config_file}'",
+            stderr=STDOUT,
+            shell=True,
+            universal_newlines=True,
+        )
+
+    except CalledProcessError as e:
+        logger.error(f"command '{e.cmd}' return with error (code {e.returncode}): {e.output}")
+        raise
+    assert f"Completed updating config for topic {topic}." in result
+
+
+def read_topic_config(model_full_name: str, app_name: str, topic: str) -> str:
+    """Helper to get a topic's configuration.
+
+    Args:
+        model_full_name: Juju model
+        app_name: Kafka app name in the Juju model
+        topic: the desired topic to read the configuration from
+    """
+    if SUBSTRATE == "k8s":
```

Comment:
**niptick** this could be factored out (since it seems just a copy and paste from above). Maybe a fixture or even just a general function, just to centralize this

Reply:
Refactored as part of 74c724d31